src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/SqlClient/TdsParserStateObject.cs

```diff
@@ -35,12 +50,6 @@ internal partial class TdsParserStateObject
         // Timeout variables
         internal bool _attentionReceived = false;               // NOTE: Received is not volatile as it is only ever accessed\modified by TryRun its callees (i.e. single threaded access)
 
-        // This variable is used to prevent sending an attention by another thread that is not the
-        // current owner of the stateObj.  I currently do not know how this can happen.  Mark added
-        // the code but does not remember either.  At some point, we need to research killing this
-        // logic.
-        private volatile int _allowObjectID;
-
         internal bool _hasOpenResult = false;
```

Comment:
Could you make private this field and use HasOpenResult property where it's not used